File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/perfil/PerfilDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.perfil;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class PerfilDTO implements DTO {
    private UUID id;
    private String nome;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
    private List<PerfilModuloDTO> permissoes;
}

File: src/frontend/src/app/components/cadastro/perfil/model/perfil-dto.ts
import { PerfilModuloDTO } from './perfil-modulo-dto';

export class PerfilDTO {
  id?: string;
  nome: string;
  createdAt?: Date;
  updatedAt?: Date;
  createdBy?: string;
  updatedBy?: string;
  permissoes: PerfilModuloDTO[];

  constructor(
    id: string | undefined,
    nome: string,
    createdAt: Date | undefined,
    updatedAt: Date | undefined,
    createdBy: string | undefined,
    updatedBy: string | undefined,
    permissoes: PerfilModuloDTO[]
  ) {
    this.id = id;
    this.nome = nome;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
    this.permissoes = permissoes;
  }
}
